ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/ActivePeers.hs

```diff
@@ -946,18 +947,19 @@ jobDemoteActivePeer
   -> peerconn
   -> Job () m (Completion m extraState extraDebugState extraFlags extraPeers peeraddr
                           peerconn)
-jobDemoteActivePeer PeerSelectionActions{peerStateActions = PeerStateActions {deactivatePeerConnection}}
+jobDemoteActivePeer PeerSelectionActions{peerStateActions = PeerStateActions {deactivatePeerConnection,
+                                                                              monitorPeerConnection}}
                     peeraddr peerconn =
     Job job handler () "demoteActivePeer"
   where
     handler :: SomeException -> m (Completion m extraState extraDebugState extraFlags extraPeers peeraddr peerconn)
-    handler e = return $
-      -- It's quite bad if demoting fails. The peer is cooling so
-      -- we can't remove it from the set of established and hot peers.
-      --
-      Completion $ \st@PeerSelectionState {
+    handler e = do
+      atomically $ monitorPeerConnection peerconn >>= check . (== PeerCold) . fst
```

Comment:
👍🏼

Reply:
add some haddocs here too, a link to the issue / PR might also be helpful.